### PR TITLE
Feature/add group contract

### PR DESCRIPTION
**Pull request by anemonekk**
### Reason for this PR
- We need groups for enabling a permission system.

### Changes in this PR
- Add Group Contract
- Add tests for Group Contract

### CODE REVIEW

File: UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/GroupContract.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+     * Gets GroupList from the ledger.
+     * @param ctx transaction context providing access to ChaincodeStub etc.
+     * @return Serialized List of Matching Groups on success, serialized error on failure
+     */
+    @Transaction()
+    public String getAllGroups(final Context ctx) {
+        ChaincodeStub stub = ctx.getStub();
+
+
+        List<Group> groupIdList = cUtil.getAllGroups(stub);
```
Comment (matthias-geuchen) [inline UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/GroupContract.java:170]:
```suggestion
        List<Group> groupList = cUtil.getAllGroups(stub);
```
This is not a group-id list. Is this intended?

File: UC4-chaincode/src/test/resources/test_configs/group_contract/AddUserToGroupTestIO.json [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+        "groupId": "0123",
+        "userList": [
+          "0000001",
+          "0000002"
+        ]
+      }
+    ]
+  },
+  {
+    "name": "addUserToTwoGroups",
```
Comment (matthias-geuchen) [inline UC4-chaincode/src/test/resources/test_configs/group_contract/AddUserToGroupTestIO.json:114]: Unfitting name. User is already in two groups, right?

File: UC4-chaincode/src/test/resources/test_configs/group_contract/AddUserToGroupTestIO.json [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+          {
+            "name": "group.enrollmentId",
+            "reason": "The user you are trying to add to a group is not registered in the system."
+          }
+        ]
+      }
+    ]
+  },
+  {
+    "name": "addNonExistingUserToEmptyGroup",
```
Comment (matthias-geuchen) [inline UC4-chaincode/src/test/resources/test_configs/group_contract/AddUserToGroupTestIO.json:194]:
```suggestion
    "name": "addUserToEmptyGroup",
```
Unfitting name. The user exists, right?

**Review by matthias-geuchen — APPROVED** [review]
Good job

**Review by mco33415 — APPROVED** [review]
See comments of Matthias.

File: UC4-chaincode/src/test/resources/test_configs/group_contract/AddUserToGroupTestIO.json (hunk 2 above)
Comment (NikoBergemann) [reply UC4-chaincode/src/test/resources/test_configs/group_contract/AddUserToGroupTestIO.json:114]:
```suggestion
    "name": "addDuplicateUserToTwoGroups",
```

File: UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/GroupContract.java [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+     * @return Serialized List of Matching Groups on success, serialized error on failure
+     */
+    @Transaction()
+    public String getAllGroups(final Context ctx) {
+        ChaincodeStub stub = ctx.getStub();
+
+
+        List<Group> groupList = cUtil.getAllGroups(stub);
+
+        return GsonWrapper.toJson(groupIdList);
```
Comment (NikoBergemann) [inline UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/GroupContract.java:172]:
```suggestion
        return GsonWrapper.toJson(groupList );
```
fix name change

**Review by NikoBergemann — APPROVED** [review]

**Review by sluis1 — APPROVED** [review]

**Pull request merged**